homework/list.js

```diff
@@ -1,79 +1,135 @@
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
+	/**
+	 *To get the data(passage) from demo-list-notes.json.
+	 *
+	 *@return {Promise} work flow
+	 *@this {TodoListManager}
+	 */
+    TodoListManager.prototype.fetchList = function(afterFetch) {
+        return new Promise((function(resolve, reject) {
+            var xhr = new XMLHttpRequest();
+            //xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
+            xhr.open('GET', 'demo-list-notes.json', true);
+            xhr.responseType = 'json';
+            xhr.onreadystatechange = function(e) {
+                // Watch out: we have a mysterious unknown 'this'.
+                if (this.readyState === 4 && this.status === 200) {
+                    var listData = this.response;
+					//The flow ends here.
+                    resolve(listData);
+                } else if (this.status !== 200) {
+                    reject('FETCHING FAILED: ' + this.status + ' ' + this.readyState);
+                }
+            };
+            xhr.send();
+        }).bind(this));
+    };
+	
+	/**
+	 *Handle the event. On click, change the content.
+	 *
+	 *@param {object} Only 'click' will happen here.
+	 *@this {TodoListManager}
+	 */
+    TodoListManager.prototype.handleEvent = function(event) {
+        switch (event.type) {
+            case 'click':
+                this.onNoteOpen(event);
+                break;
+        }
     }
-  }
-
-  function updateList(list) {
-    _listNoteContent = list;
-  }
-
-  function drawList() {
-    var list = _listNoteContent;
-    var ul = document.createElement('ul');
-    ul.id = 'note-title-list';
-    var buff = document.createDocumentFragment();
-    list.forEach(function(note, i) {
-      var li = document.createElement('li');
-      li.dataset.noteId = i;
-      li.classList.add('note-title');
-      li.textContent = note.title;
-      // Note: buff is captured, so we now have a
-      // little closure naturally.
-      buff.appendChild(li);
-    });
-    ul.appendChild(buff);
-    _wrapper.appendChild(ul);
-  }
-
-  function fetchList(afterFetch) {
-    var xhr = new XMLHttpRequest();
-    xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
-    xhr.responseType = 'json';
-    xhr.onreadystatechange = function(e) {
-      // Watch out: we have a mysterious unknown 'this'.
-      if (this.readyState === 4 && this.status === 200) {
-        var listData = this.response;
-        // The flow ends here.
-        afterFetch(listData);
-      } else if (this.status !== 200 ){
-        // Ignore error in this case.
-      }
+		
+	/**
+	 *Start here. get the list id.
+	 *
+	 *@this {TodoListManager}
+	 */
+    TodoListManager.prototype.start = function() {
+        this._wrapper = document.querySelector('#note-list-wrapper')
+        this.fetchList()
+            .then((function(data) {
```

Comment:
You should practice to compose a chain with multiple `then` and `catch`. And you should note that `catch` must come with `logging AND re-throw`, or the following chains may screw you, if there are any error in the chain.